### ISSUE TITLE
About pretrained model

### ISSUE THREAD

**Issue by Ninacheng222** [salu133445/musegan#20]
1. About the pretrained model
I use the pretrained model `lastfm_alternative_g_hybrid_d_proposed`, but the structure seems to be different from the code structure so that the pretrained model cannot be used.

> Caused` by op 'GAN/save/RestoreV2', defined at:
>   File "main.py", line 125, in <module>
>     main()
>   File "main.py", line 57, in main
>     gan = MODELS.GAN(sess, CONFIG['model'])
>   File "/home/nina.cheng/musegan/musegan/musegan/models.py", line 19, in __init__
>     self.build()
>   File "/home/nina.cheng/musegan/musegan/musegan/models.py", line 92, in build
>     self.saver = tf.train.Saver()
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/training/saver.py", line 1284, in __init__
>     self.build()
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/training/saver.py", line 1296, in build
>     self._build(self._filename, build_save=True, build_restore=True)
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/training/saver.py", line 1333, in _build
>     build_save=build_save, build_restore=build_restore)
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/training/saver.py", line 781, in _build_internal
>     restore_sequentially, reshape)
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/training/saver.py", line 400, in _AddRestoreOps
>     restore_sequentially)
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/training/saver.py", line 832, in bulk_restore
>     return io_ops.restore_v2(filename_tensor, names, slices, dtypes)
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/ops/gen_io_ops.py", line 1463, in restore_v2
>     shape_and_slices=shape_and_slices, dtypes=dtypes, name=name)
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
>     op_def=op_def)
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3414, in create_op
>     op_def=op_def)
>   File "/home/nina.cheng/.pyenv/versions/py3_tensorflow/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1740, in __init__
>     self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access
> 
> NotFoundError (see above for traceback): Key GAN/G/bar_main_0/Layer_1/batch_norm/beta not found in checkpoint
> 	 [[Node: GAN/save/RestoreV2 = RestoreV2[dtypes=[DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, ..., DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_INT32], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_GAN/save/Const_0_0, GAN/save/RestoreV2/tensor_names, GAN/save/RestoreV2/shape_and_slices)]]
> 	 [[Node: GAN/save/RestoreV2/_169 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_170_GAN/save/RestoreV2", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

2. About training warning
When I training the model, it has some warning like this:

> /home/nina.cheng/musegan/musegan/utils/metrics.py:92: RuntimeWarning: invalid value encountered in true_divide
>   chroma1 = chroma1 / np.sum(chroma1)
> /home/nina.cheng/musegan/musegan/utils/metrics.py:94: RuntimeWarning: invalid value encountered in true_divide
>   chroma2 = chroma2 / np.sum(chroma2)
> /home/nina.cheng/musegan/musegan/utils/metrics.py:92: RuntimeWarning: invalid value encountered in true_divide
>   chroma1 = chroma1 / np.sum(chroma1)
> /home/nina.cheng/musegan/musegan/utils/metrics.py:94: RuntimeWarning: invalid value encountered in true_divide
>   chroma2 = chroma2 / np.sum(chroma2)
What does it means? Does it matters?

**Comment by salu133445** [comment]
For the second issue, it doesn't matter. All the NAN values will be later replaced by zeros when we compute the metric values, so no worries. You can suppress these numpy warnings by:

```python
import numpy as np
np.seterr(divide='ignore', invalid='ignore')
```

**Comment by salu133445** [comment]
We have updated the code to v3. It should now be easier to load the pretrained models.
